aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -415,17 +415,31 @@ where
             match change {
                 AggregatorChange::Write(value) => {
                     let bytes = serialize(&value);
-                    let move_op = if write_set_mut.get(&state_key).is_none() {
-                        MoveStorageOp::New(bytes)
-                    } else {
-                        MoveStorageOp::Modify(bytes)
+                    let move_op = match write_set_mut.get(&state_key) {
+                        None => MoveStorageOp::New(bytes),
+                        Some(direct_write) => match direct_write {
+                            WriteOp::Creation(_) | WriteOp::CreationWithMetadata { .. } => {
+                                MoveStorageOp::New(bytes)
+                            },
+                            // FIXME(alden): is this an error?
```

Comment:
@georgemitenkov  @gelash 
1. Is it true that when an aggregator is first created, there will be a Create from the table change set and then a modify here from the aggregator change set?
2. is that the only known case to you wrt aggregators that a state key is repeated in the (outer) change set? 

Any other concerns?